server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -634,6 +641,14 @@ public void run() {
         }
     }
 
+    private void cancelTask(SearchTask task, Exception exc) {
+        CancelTasksRequest req = new CancelTasksRequest()
+            .setTaskId(new TaskId(client.getLocalNodeId(), task.getId()))
+            .setReason(exc.getMessage());
```

Comment:
Maybe add some prefix to the message? I am always afraid that the exc message may be null or not so understandable, with a prefix we would know for sure where the cancel comes from.